src/libraries/System.Threading.RateLimiting/tests/PartitionedRateLimiterTests.cs

```diff
@@ -403,6 +401,36 @@ public async Task Create_CancellationTokenPassedToUnderlyingLimiter()
             await Assert.ThrowsAsync<TaskCanceledException>(async () => await waitTask);
         }
 
+        [Fact]
+        public async Task IdleLimiterIsCleanedUp()
+        {
+            ExpiredLimiter expiredLimiter = null;
+            var factoryCallCount = 0;
+            using var limiter = PartitionedRateLimiter.Create<string, int>(resource =>
+            {
+                return RateLimitPartition.Create(1, _ =>
+                {
+                    factoryCallCount++;
+                    expiredLimiter = new ExpiredLimiter();
+                    return expiredLimiter;
+                });
+            });
+
+            var lease = limiter.Acquire("");
+            Assert.True(lease.IsAcquired);
+
+            Assert.Equal(1, factoryCallCount);
+
+            expiredLimiter.IdleLength = TimeSpan.FromMinutes(1);
+            // Limiter is disposed when timer runs and sees that IdleDuration is greater than idle limit
+            await expiredLimiter.DisposeCts.Task;
```

Comment:
Would it be cleaner and faster to call `limiter.TimerLoop()` using private reflection?

Reply:
Faster, but not good since we currently assume the TimerLoop is only called one at time and this could violate that

Reply:
Can we disable the timer by calling an internal version of `PartitionedRateLimiter.Create` using private reflection? I hate having the test rely on an actual timer that only fires 10 times a second.